analyzers/src/SonarAnalyzer.CSharp/Wrappers/ObjectCreationFactory.cs

```diff
@@ -0,0 +1,77 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Wrappers
+{
+    public interface IObjectCreation
+    {
+        InitializerExpressionSyntax Initializer { get; }
+        ArgumentListSyntax ArgumentList { get; }
+        ExpressionSyntax Expression { get; }
+        string TypeAsString(SemanticModel semanticModel);
+    }
+
+    public class ObjectCreationFactory
+    {
+        public static IObjectCreation Create(SyntaxNode node) =>
+            node switch
+            {
+                null => throw new ArgumentNullException(nameof(node), "Argument should not be null"),
```

Comment:
This will work just as fine, as there's nothing additional to be said in the error
```suggestion
                null => throw new ArgumentNullException(nameof(node)),
```